### PR TITLE
Also build for macos arm

### PR DESCRIPTION
**Pull request by janjagusch**
Checklist
* [x] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [x] Bumped the build number (if the version is unchanged)
* [ ] Reset the build number to `0` (if the version changed)
* [ ] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [ ] Ensured the license file is being packaged.

### CODE REVIEW

**Comment by janjagusch** [comment]
@conda-forge-admin, please rerender

**Comment by janjagusch** [comment]
@xhochy i'm trying to make this feedstock compatible with osx arm64. i don't really know what i'm doing, though.
i've been following ["How to add a osx-arm64 build to a feedstock" documentation](https://conda-forge.org/blog/posts/2020-10-29-macos-arm64/) but it doesn't really explain how to make it work with a yarn package.

currently i'm getting this error:

`/Users/runner/miniforge3/conda-bld/prettier_1631724400678/_h_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_/bin/yarn: line 20: /Users/runner/miniforge3/conda-bld/prettier_1631724400678/_h_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_/bin/node: Bad CPU type in executable`

i couldn't find any issues that would indicate that you can't build prettier for osx arm64, so i'm probably doing something stupid.

could you maybe take a look and help me out? thanks. 🙂

**Comment by xhochy** [comment]
Can you try adding `nodejs` to `build:`, and then adding 

```
rm $PREFIX/bin/node
ln -s $BUILD_PREFIX/bin/node $PREFIX/bin/node
```

before the calls to `yarn`?

**Comment by janjagusch** [comment]
> Can you try adding `nodejs` to `build:`, and then adding
> 
> ```
> rm $PREFIX/bin/node
> ln -s $BUILD_PREFIX/bin/node $PREFIX/bin/node
> ```
> 
> before the calls to `yarn`?

Thanks, now the nodejs 16 build is succeeding. For 12 and 14 there doesn't seem to be a osx arm64 compatible build. What should we do about those?

**Comment by xhochy** [comment]
> Thanks, now the nodejs 16 build is succeeding. 

Can you add `store_build_artifacts: true` (see https://conda-forge.org/docs/maintainer/conda_forge_yml.html#azure-config) to the config? We don't run package tests for `osx-arm64` on CI as there isn't any emulation available. As this is the first cross-compiled `nodejs` package, I would be very careful with merging and first test the result locally.

> For 12 and 14 there doesn't seem to be a osx arm64 compatible build. What should we do about those?

These should not be part of the pinning, can you add a `# [not (osx and arm64)]` to the global pinning here: https://github.com/conda-forge/conda-forge-pinning-feedstock/blob/7019505995a030f45a267db3f7a71c964b9e5556/recipe/conda_build_config.yaml#L567-L570

**Comment by janjagusch** [comment]
> > Thanks, now the nodejs 16 build is succeeding.
> 
> Can you add `store_build_artifacts: true` (see https://conda-forge.org/docs/maintainer/conda_forge_yml.html#azure-config) to the config? We don't run package tests for `osx-arm64` on CI as there isn't any emulation available. As this is the first cross-compiled `nodejs` package, I would be very careful with merging and first test the result locally.
> 
> > For 12 and 14 there doesn't seem to be a osx arm64 compatible build. What should we do about those?
> 
> These should not be part of the pinning, can you add a `# [not (osx and arm64)]` to the global pinning here: https://github.com/conda-forge/conda-forge-pinning-feedstock/blob/7019505995a030f45a267db3f7a71c964b9e5556/recipe/conda_build_config.yaml#L567-L570

Done.

https://github.com/conda-forge/conda-forge-pinning-feedstock/pull/1900

**Comment by janjagusch** [comment]
@conda-forge-admin, please restart ci.

**Comment by xhochy** [comment]
You need @conda-forge-admin please rerender once you have updated the pinning

**Comment by xhochy** [comment]
@janjagusch Feel free to ping me on Monday to test this!

**Comment by janjagusch** [comment]
> @janjagusch Feel free to ping me on Monday to test this!

If you need my help to test this then we might have to move this to tomorrow. Otherwise: *ping*! 🙂

File: recipe/build.sh [hunk 1]
```diff
@@ -1,4 +1,15 @@
+#!/usr/bin/env bash
+
+echo "$target_platform"
+if [ "$target_platform" == "osx-arm64" ]; then
+    export npm_config_arch="x64"
+fi
```
Comment (xhochy) [inline recipe/build.sh:6]:
```suggestion
if [[ "${target_platform}" == "osx-arm64" ]]; then
    export npm_config_arch="arm64"
fi
```

File: recipe/build.sh [hunk 2]
```diff
@@ -1,4 +1,15 @@
+#!/usr/bin/env bash
+
+echo "$target_platform"
```
Comment (xhochy) [inline recipe/build.sh:3]:
```suggestion
set -euxo pipefail
```

**Pull request merged**